lib/src/Uncertainty/Distribution/TruncatedDistribution.cxx

```diff
@@ -414,31 +454,34 @@ Scalar TruncatedDistribution::getThresholdRealization() const
 /* Lower bound accessor */
 void TruncatedDistribution::setLowerBound(const Scalar lowerBound)
 {
-  if ((finiteUpperBound_) && (lowerBound > upperBound_)) throw InvalidArgumentException(HERE) << "Error: the lower bound must be strictly less than the upper bound, here lower bound=" << lowerBound << " and upper bound=" << upperBound_;
+  LOGWARN("TruncatedDistribution::setLowerBound is deprecated, use setBound..");
+  if ((bounds_.getFiniteUpperBound()[0]) && (lowerBound > bounds_.getUpperBound()[0])) throw InvalidArgumentException(HERE) << "Error: the lower bound must be strictly less than the upper bound, here lower bound=" << lowerBound << " and upper bound=" << bounds_.getUpperBound()[0];
   cdfLowerBound_ = distribution_.computeCDF(Point(1, lowerBound));
   if (cdfLowerBound_ >= cdfUpperBound_) throw InvalidArgumentException(HERE) << "Error: the truncation interval does not contain a non-empty part of the support of the distribution";
   pdfLowerBound_ = distribution_.computePDF(Point(1, lowerBound));
-  lowerBound_ = lowerBound;
-  finiteLowerBound_ = true;
+  bounds_.setLowerBound(Point(1, lowerBound));
+  bounds_.setFiniteLowerBound(BoolCollection(1, true));
   normalizationFactor_ = 1.0 / (cdfUpperBound_ - cdfLowerBound_);
   computeRange();
 }
 
 Scalar TruncatedDistribution::getLowerBound() const
 {
-  return lowerBound_;
+  LOGWARN("TruncatedDistribution::getLowerBound is deprecated, use getRange.");
```

Comment:
No. Either we call getBound() if we want to keep track of the bound given by the user (ie before its restriction wrt the range of the underlying distribution) or we call getRange() but use range_ everywhere in the code.